test/config/utility.cc

```diff
@@ -844,6 +844,23 @@ void ConfigHelper::finalize(const std::vector<uint32_t>& ports) {
 
   applyConfigModifiers();
 
+  setPorts(ports);
+
+  if (!connect_timeout_set_) {
```

Comment:
i'm curious why this was moved from its prior position a dozen lines down -- is this because you factored out a big function into some smaller ones but the order in the file no longer matches the order of execution?

Reply:
yeah I moved out the set ports code and the diffs are weird.